clients/benchmarks/client.cpp

```diff
@@ -113,6 +109,27 @@ struct perf_blas : rocblas_test_invalid
 {
 };
 
+template <typename T>
+struct perf_blas<T,
+                 typename std::enable_if<std::is_same<T, rocblas_float_complex>{}
+                                         || std::is_same<T, rocblas_double_complex>{}>::type>
+{
+    explicit operator bool()
+    {
+        return true;
+    }
+    void operator()(const Arguments& arg)
+    {
+        if(!strcmp(arg.function, "gemm"))
+            testing_gemm<T>(arg);
+        else if(!strcmp(arg.function, "gemm_strided_batched"))
+            testing_gemm_strided_batched<T>(arg);
+        else
+            throw std::invalid_argument("Invalid combination --function "s + arg.function
+                                        + " --a_type "s + rocblas_datatype2string(arg.a_type));
+    }
+};
+
```

Comment:
Rather than write a separate specialization for complex, the `enable_if` for the current real `gemm` should be expanded to support complex by adding `|| is_complex<T>` to its enabling condition. Then you do not need to duplicate code.

Reply:
Yes, I understand. But most of our function, like trsm, didn't support complex. For preventing use complex in other function, we prefer to write another specialization to handle complex. This is same as how we handle half type.